### ISSUE TITLE
Setting DPI too high causes browser render to be cut off

### ISSUE THREAD

**Issue by Neriderc** [Neriderc/GVExport#138]
When you set the DPI setting higher than 72, the right and bottom edges get cut off.

We should see what options are, e.g.:
- If this can be fixed, fix it
- If this can't be fixed (e.g. if it's outside our control), then use fixed 72DPI setting for browser and have setting only apply to server side download.

**Comment by Neriderc** [comment]
Fixed this by adjusting the view port to the size of the image - so it does affect client side (i.e. the first option above, fixing it). It's worth noting that the browser has a max canvas size (limited by number of pixels), and canvas is needed for exporting JPG/PNG. If you set DPI too high, you will hit the limit and the canvas comes out blank. I have added code to detect this and instead of downloading a blank file, it will show an error message (toast). This is a browser limitation, not a bug.
